### PR TITLE
Add delivery report to Balsamic-QC workflow

### PR DESCRIPTION
**Pull request by ivadym**
### Added
- Delivery report to Balsamic-QC workflow

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [x] `cg generate delivery-report <qc-only-case-id>`

### Expected test outcome

- [x] Delivery report is generated
Screenshot 2023-11-20 at 17 45 00

- [x] Delivery report is delivered to customer inbox
Screenshot 2023-11-20 at 17 49 54

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/constants/delivery.py [hunk 1]
```diff
@@ -42,6 +42,7 @@
 ]
 
 BALSAMIC_QC_ANALYSIS_CASE_TAGS = [
```
Comment (henrikstranneheim) [inline cg/constants/delivery.py:44]: Add type hint

File: cg/constants/report.py [hunk 2]
```diff
@@ -71,23 +71,23 @@
 ]
 
 # Data analysis required fields
-_REQUIRED_DATA_ANALYSIS_FIELDS = [
+REQUIRED_DATA_ANALYSIS_FIELDS = [
```
Comment (henrikstranneheim) [inline cg/constants/report.py:74]: Add type hint

File: cg/constants/report.py [hunk 3]
```diff
@@ -71,23 +71,23 @@
...
     "customer_pipeline",
     "pipeline",
     "pipeline_version",
     "genome_build",
 ]
 
-REQUIRED_DATA_ANALYSIS_MIP_DNA_FIELDS = _REQUIRED_DATA_ANALYSIS_FIELDS + [
+REQUIRED_DATA_ANALYSIS_MIP_DNA_FIELDS = REQUIRED_DATA_ANALYSIS_FIELDS + [
```
Comment (henrikstranneheim) [inline cg/constants/report.py:81]: Type hints

File: cg/meta/report/balsamic.py [hunk 4]
```diff
@@ -174,6 +175,11 @@ def get_report_accreditation(
     def get_required_fields(self, case: CaseModel) -> dict:
         """Retrieves a dictionary with the delivery report required fields for BALSAMIC."""
```
Comment (henrikstranneheim) [inline cg/meta/report/balsamic.py:176]:
```suggestion
        """Retirn a dictionary with the delivery report required fields for BALSAMIC."""
```

**Review by henrikstranneheim — APPROVED** [review]
Great 👍

**Comment by ivadym** [comment]
Screenshot 2023-11-21 at 10 16 57

**Pull request merged**